src/components/accordion/accordion.tsx

```diff
@@ -22,32 +23,64 @@ export const Accordion = ({ items, className }: AccordionProps) => {
             {items.map((item, index) => {
                 const isOpen = openItemIndex === index;
                 return (
-                    <div key={index} className={styles.item}>
-                        <div
-                            className={styles.header}
-                            {...getClickableElementAttributes(() =>
-                                setOpenItemIndex(isOpen ? null : index),
-                            )}
-                        >
-                            <p className={styles.title}>{item.title}</p>
-
-                            {isOpen ? (
-                                <MinusIcon className={styles.toggleIcon} />
-                            ) : (
-                                <PlusIcon className={styles.toggleIcon} />
-                            )}
-                        </div>
-
-                        <div
-                            className={classNames(styles.content, {
-                                [styles.contentVisible]: isOpen,
-                            })}
-                        >
-                            <div className={styles.contentInner}>{item.content}</div>
-                        </div>
-                    </div>
+                    <AccordionItem
+                        key={index}
+                        isOpen={isOpen}
+                        onToggle={() => setOpenItemIndex(isOpen ? null : index)}
+                        {...item}
+                    />
                 );
             })}
         </div>
     );
 };
+
+interface AccordionItemProps extends AccordionItem {
+    isOpen: boolean;
+    onToggle: () => void;
+}
+
+const AccordionItem = ({ isOpen, onToggle, title, content }: AccordionItemProps) => {
+    const contentRef = useRef<HTMLDivElement | null>(null);
+
+    const isOpenRef = useRef(isOpen);
```

Comment:
It doesn't look like you need `isOpenRef`, since it only changes if the prop changes, and `handleAnimationComplete` will be regenerated in that case.

Or does `motion.div` ignore changes to the `onAnimationComplete` prop?

Reply:
Yes, `motion.div` ignores the change so if I use `isOpen` it's not up to date.

Reply:
🥲👍